### ISSUE TITLE
Is it possible to adjust this framework(slim folder) to RNN?

### ISSUE THREAD

**Issue by guotong1988** [tensorflow/models#1076]
I read that [here](https://github.com/tensorflow/models/blob/master/slim/train_image_classifier.py#L555)(or from [here](https://github.com/tensorflow/models/blob/master/slim/train_image_classifier.py#L400) to [here](https://github.com/tensorflow/models/blob/master/slim/train_image_classifier.py#L570))is maybe a optimized code for multi-gpu training.

So, can it also be used to RNN?

If it can also be used to RNN , I want to try to edit it (for myself).

Or is there any data-parallel framework for RNN?

**Comment by guotong1988** [comment]
Thank you. @yaroslavvb @nealwu @cshallue @martinwicke @panyx0718 @calberti @lukaszkaiser @jart @nathansilberman @aselle @mrry

**Comment by lukaszkaiser** [comment]
I think model_deploy (or any form of data parallelism) is universal. It usually only asumes your batch axis is axis=0, then it can split, run your model, and merge. So it should work with RNNs same as with any other model. Please feel free to reopen this issue if you have trouble with that!

**Comment by guotong1988** [comment]
@lukaszkaiser I haven't seen the reopen button.

Thank you for your answer and then my question is 

Do you think it will be a speed-up if we use it for RNN?

**Comment by lukaszkaiser** [comment]
It is hard to tell. It depends on your batch size, LSTM cell size, maybe even on the LSTM architecture. The easiest way is to try it out.

**Comment by guotong1988** [comment]
@lukaszkaiser Thank you.
I have change the slim folder to a multi-machine code for data parallelism training.
And I will try it.
Do you think it is different from the normal between-graph [way](https://www.tensorflow.org/versions/r0.11/how_tos/distributed/) which use `tf.train.Supervisor` and `tf.train.replica_device_setter`
